include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2022  Andrea Bocci, Mehmet Yusufoglu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+//! ALPAKA_DEVICE_THROW either aborts(terminating the program and creating a core dump) or throws std::runtime_error
+//! depending on the Acc. The std::runtime_error exception can be catched in the catch block.
+//!
+//! CUDA __trap function triggers std::runtime_error but can be catched during wait not exec.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_DEVICE_THROW aborts the the program for OpenMP backends. If needed
+//! the SIGABRT signal can be catched by signal handler.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
```

Comment:
I was on the way to say `change it to a C++ function and than this should be `deviceThrow()` within the alpaka namespace.`
**BUT** than it should be a trait which gets the acc as argument with the coresponding class equal to the trait name which is than specialized for each acc. Device functions always take the acc as first argument.

Never the less I would handle it equal to `ALPAKA_ASSERT_ACC` and would add the line and error message to the printf.
To follow the naming schema this macro should be named `ALPAKA_THROW_ACC`

Reply:
Hence; according to you it should be a macro called ALPAKA_THROW_ACC without an Acc argument?

Reply:
yes this will follow our macro `ALPAKA_ASSERT_ACC` and avoid introducing a new trait. IMO this function/macro will be used in seldem cases therefore I would like to avoid increasing the code base because of this tiny throw.

Reply:
macro name changed to ALPAKA_THROW_ACC. Thanks.